src/main/kotlin/no/skatteetaten/aurora/mokey/service/ManagementEndpoint.kt

```diff
@@ -25,88 +27,150 @@ import org.springframework.web.client.RestTemplate
 import kotlin.reflect.KClass
 
 @Service
-class ManagementEndpointFactory(val restTemplate: RestTemplate) {
-    fun create(managementUrl: String): ManagementEndpoint {
-        return ManagementEndpoint.create(restTemplate, managementUrl)
+class ManagementInterfaceFactory(val restTemplate: RestTemplate) {
+    fun create(discoveryUrl: String): Pair<ManagementInterface?, ManagementEndpointResult<ManagementLinks>> {
+        return ManagementInterface.create(restTemplate, discoveryUrl)
     }
 }
 
-class ManagementEndpointException(
-    val endpoint: Endpoint,
-    val errorCode: String,
-    url: String? = null,
-    cause: Exception? = null
-) : RuntimeException("${endpoint}_$errorCode", cause)
-
-class ManagementEndpoint internal constructor(
-    private val restTemplate: RestTemplate,
-    val links: ManagementLinks
-) {
-
-    @Throws(ManagementEndpointException::class)
-    fun getHealthEndpointResponse(): HttpResponse<HealthResponse> {
-        val response = findJsonResource(HEALTH, JsonNode::class)
-        val healthResponse = HealthResponseParser.parse(response.deserialized)
-        return HttpResponse(healthResponse, response.textResponse, response.createdAt)
+class ManagementEndpoint(val url: String, private val endpointType: Endpoint) {
+
+    val logger: Logger = LoggerFactory.getLogger(ManagementEndpoint::class.java)
+
+    fun <T : Any> findJsonResource(restTemplate: RestTemplate, clazz: KClass<T>): ManagementEndpointResult<T> {
+        logger.debug("Getting resource with url={}", url)
+
+        try {
+            val responseText: String = try {
+                restTemplate.getForObject(url, String::class.java)
+            } catch (e: HttpStatusCodeException) {
+                if (!e.statusCode.is5xxServerError) throw e
+                String(e.responseBodyAsByteArray)
+            } ?: "Error while communicating with management endpoint"
```

Comment:
Jeg er litt usikker på om dette er riktig (og litt vanskelig å skjønne med en gang). Denne fallbacken for null peker egentlig på restTemplate.getForObject(url, String::class.java), vel? Når returnerer getForObject null? Jeg tror det kan være misvisende å si at dette er "Error while communicating with management endpoint".

Reply:
Denne metoden er omstrukturert iht tilbakemelding.